staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +652,62 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
+	replaceKeysInPatch, foundInPatch := patch[replaceKeysDirective]
+	delete(patch, replaceKeysDirective)
+	if !options.MergeParallelList {
+		replaceKeysInOriginal, foundInOriginal := original[replaceKeysDirective]
+		switch {
+		case foundInOriginal && foundInPatch:
+			if !reflect.DeepEqual(replaceKeysInOriginal, replaceKeysInPatch) {
+				return fmt.Errorf("%v and %v are not deep equal", replaceKeysInOriginal, replaceKeysInPatch)
+			}
+		case !foundInOriginal && foundInPatch:
+			original[replaceKeysDirective] = replaceKeysInPatch
+		}
+		return nil
+	}
+
+	if !foundInPatch {
+		return nil
+	}
+	replaceKeysList, ok := replaceKeysInPatch.([]interface{})
+	if !ok {
+		return mergepatch.ErrBadPatchFormatForReplaceKeys
+	}
+
+	// validate patch to make sure all field in patch is covered in replaceKeysList
+	m := map[interface{}]bool{}
```

Comment:
For what it's worth, you can use the empty struct `struct{}` as an indicator of presence. I've seen a lot of Go programmers do this to avoid allocating space for a bunch of redundant booleans.